src/www/SQLiteParamAdapter.ts

```diff
@@ -0,0 +1,149 @@
+     * convered to an integer of 1 or 0, Dates are converted to an ISO string,
+     * and blobs are adapted to a special JSON format to represent bytearrays.
+     * 
+     * If required, any of the _adapt* methods could be overwritten to change
+     * the behaviour however it would be recommended to keep the defaults.
+     * 
+     * If you are importing custom types, override the `_adapt` method to handle
+     * them.
+     * 
+     * @param v 
+     * @returns 
+     */
+    private async $adapt(v: unknown): Promise<SQLiteType> {
+        let out: SQLiteType;
+        if (v === null) {
+            out = this._adaptNull(<null>v);
+        }
+        else if (typeof v === 'number') {
+            out = this._adaptNumber(v);
+        }
+        else if (typeof v === 'string') {
+            out = this._adaptString(v);
+        }
+        else if (typeof v === 'boolean') {
+            out = this._adaptBoolean(v);
+        }
+        else if (v instanceof Date) {
+            out = this._adaptDate(v);
+        }
+        else if (v instanceof Blob) {
+            out = await this._adaptBlob(v);
+        }
+        else if (v instanceof ArrayBuffer) {
+            out = await this._adaptArrayBuffer(v);
+        }
+        else if ((v instanceof Int8Array) || (v instanceof Uint8Array)) {
+            out = await this._adaptInt8TypedArray(v);
+        }
+        else {
+            out = await this._adapt(v);
+        }
+
+        return out;
+    }
+
+    protected async _adaptInt8TypedArray(v: Int8Array | Uint8Array): Promise<SQLiteType> {
+        return await SQLiteParamValueConverter.int8OrUint8ToSQLiteBlob(v);
+    }
+
+    protected async _adaptArrayBuffer(v: ArrayBuffer): Promise<SQLiteType> {
+        return await SQLiteParamValueConverter.arrayBufferToSQLiteBlob(v);
+    }
+
+    protected async _adaptBlob(v: Blob): Promise<SQLiteType> {
+        return await SQLiteParamValueConverter.blobToSQLiteBlob(v);
+    }
+
+    protected _adaptDate(v: Date): SQLiteType {
+        return SQLiteParamValueConverter.dateToText(v);
+    }
+
+    protected _adaptBoolean(v: boolean): SQLiteType {
+        return SQLiteParamValueConverter.booleanToInteger(v);
+    }
+
+    protected _adaptString(v: string): SQLiteType {
+        return v;
+    }
+
+    protected _adaptNumber(v: number): SQLiteType {
+        return v;
+    }
+
+    protected _adaptNull(v: null): SQLiteType {
+        return null;
+    }
+
+    protected async _adapt(v: unknown): Promise<SQLiteType> {
+        console.warn('Passing through an unknown type. This is potentially dangerous and an unsupported operation.\nTo support custom types, supply a SQLiteParamAdapter that has the knowledge to convert custom types into one of the SQLite Types.');
+        return <SQLiteType>v;
+    }
+
+    public async processArray(input: unknown[]): Promise<SQLiteParams> {
+        // any cause we need to match the array structure...
+        let out: any = [];
```

Comment:
No, this is wrong and the types are incorrect.
You're returning an Array of data but the return type says Record of data. There's a type mismatch.

The comment doesn't make any sense. There is no matching of array structure, just use the sqlite array type. SQLiteArray | SQLiteListArgs.
Also fix the return type.